alignment/en.py

```diff
@@ -0,0 +1,303 @@
+from collections import deque
+import itertools
+from datasets import load_dataset
+import re
+import datetime
+
+now_timer = datetime.datetime.now()
+
+"""
+工作方向：
+1. 先做英语的段落。
+2. 把其它语言的段落映射到英语去。
+
+
+"""
+
+dataset = load_dataset("ranWang/test_pdf_data", split='new')
+
+filtered_file = []
+
+EDIT_DISTANCE_THRESOLD = 3
+
+def edit_distance(s1, s2):
+    """chatgpt帮我写的n方编辑距离算法"""
+    if len(s1) > len(s2):
+        s1, s2 = s2, s1
+
+    if len(s2) - len(s1) > EDIT_DISTANCE_THRESOLD: # 优化相当大的O(1)剪枝
+        return EDIT_DISTANCE_THRESOLD + 1
+    
+    # O(n)统计字符，进一步剪掉一些不必要用n^2编辑距离的情况 625s优化到22s
+    char_distance = 0
+    d = {}
+    for s in s1:
+        d[s] = d.get(s, 0) + 1
+    for s in s2:
+        d[s] = d.get(s, 0) - 1
+    positive = 0
+    negative = 0
+    for k, v in d.items():
+        if v > 0:
+            positive += v
+        else:
+            negative += - v
+    char_distance = max(positive, negative)
+    if char_distance > EDIT_DISTANCE_THRESOLD:
+        return char_distance
+
+    distances = range(len(s1) + 1)
+
+    for i2, c2 in enumerate(s2):
+        distances_ = [i2+1]
+        for i1, c1 in enumerate(s1):
+            if c1 == c2:
+                distances_.append(distances[i1])
+            else:
+                distances_.append(1 + min((distances[i1], distances[i1 + 1], distances_[-1])))
+        distances = distances_
+
+    return distances[-1]
+
+def read_int(s: str) -> int:
+    "从开头开始读一段连续的数字"
+    x = 0
+    for c in s:
+        if c.isdigit():
+            x = x * 10 + int(c)
+        else:
+            return x
+    return x
+
+truncate_token = 'This record contains the text of speeches delivered in English and of the '
```

Comment:
- global variable should be capitalized
- try to put all global variable in one place